src/main/java/nl/workingtalent/wtacademy/user/UserController.java

```diff
@@ -28,53 +27,60 @@ public class UserController {
 	public ResponseDto findAllUsers() {
 		List<User> users = service.findAllUsers();
 
-		return createResponseDtoList(null, users, null);
-	}
+		if (users.isEmpty()) {
+			ResponseDto responseDto = new ResponseDto(false, null, null, "No users found.");
 
-	@RequestMapping("user/{id}")
-	public ResponseDto findUserById(@PathVariable("id") long id) {
-		Optional<User> userOptional = service.findUserById(id);
+			return responseDto;
+		}
 
-		return createResponseDto(id, userOptional, "id");
-	}
+		Stream<ReadUserDto> readUserDtoStream = users.stream().map(user -> {
+			return new ReadUserDto(user);
+		});
 
-	@RequestMapping("user/firstname/{firstName}")
-	public ResponseDto findAllUsersByFirstName(@PathVariable("firstName") String name) {
-		List<User> users = service.findUserByFirstName(name);
+		ResponseDto responseDto = new ResponseDto(true, readUserDtoStream, null, users.size() + " users " + " found.");
 
-		return createResponseDtoList(name, users, "first name");
+		return responseDto;
 	}
 
-	@RequestMapping("user/lastname/{lastName}")
-	public ResponseDto findUserByLastName(@PathVariable("lastName") String name) {
-		List<User> users = service.findUserByLastName(name);
+	@RequestMapping("user/{id}")
+	public ResponseDto findUserById(@PathVariable("id") long id) {
+		Optional<User> userOptional = service.findUserById(id);
 
-		return createResponseDtoList(name, users, "last name");
+		if (userOptional.isPresent()) {
+			User user = userOptional.get();
+			ReadUserDto readUserDto = new ReadUserDto(user);
+			ResponseDto responseDto = new ResponseDto(true, readUserDto, null, "User found.");
+			return responseDto;
+		}
 
-	}
+		ResponseDto responseDto = new ResponseDto(false, null, null, "No user with id '" + id + "' found.");
 
-	@RequestMapping("user/email/{email}")
-	public ResponseDto findUserByEmail(@PathVariable("email") String email) {
-		Optional<User> userOptional = service.findUserByEmail(email);
+		return responseDto;
 
-		return createResponseDto(email, userOptional, "email");
 	}
 
-	@RequestMapping("user/role/{role}")
-	public ResponseDto findUserByRole(@PathVariable("role") Role role) {
-		List<User> users = service.findUserByRole(role);
+	@RequestMapping("user/search")
+	public Stream<ReadUserDto> searchUser(@RequestBody SearchUserDto dto) {
 
-		return createResponseDtoList(role, users, "role");
+		Stream<ReadUserDto> dtos = service.searchUser(dto).stream().map((user) -> {
+			return new ReadUserDto(user);
+		});
+
+		return dtos;
 	}
```

Comment:
Zou ook nog kunnen zoeken op users met bijvoorbeeld minimaal 1 review of  maximaal 5 leningen. Op dit moment nog niet belangrijk.

Reply:
Mooie feature. Zouden wij kunnen implementeren als er nog wat tijd over is